mqtt/mqttd/src/broker/bootstrap/edgehub.rs

```diff
@@ -32,38 +53,38 @@ pub async fn broker(
 }
 
 pub async fn start_server<Z, F>(
-    config: &BrokerConfig,
+    config: &Settings,
     broker: Broker<Z>,
     shutdown_signal: F,
 ) -> Result<BrokerSnapshot>
 where
     Z: Authorizer + Send + 'static,
     F: Future<Output = ()> + Unpin,
 {
-    // TODO read from config
-    let url = "http://localhost:7120/authenticate/".into();
-    let authenticator = EdgeHubAuthenticator::new(url);
-
     let mut server = Server::from_broker(broker);
 
-    if let Some(tcp) = config.transports().tcp() {
+    // Add system transport to allow communication between edgehub components
+    let authenticator = LocalAuthenticator::new();
+    server.tcp(config.listener().system().addr(), authenticator);
+
+    // Add regular MQTT over TCP transport
+    let authenticator = EdgeHubAuthenticator::new(config.auth().url());
+    if let Some(tcp) = config.listener().tcp() {
         server.tcp(tcp.addr(), authenticator.clone());
     }
 
-    let renewal_signal = match config.transports().tls() {
+    // Add regular MQTT over TLS transport
+    let renewal_signal = match config.listener().tls() {
         Some(tls) => {
-            let identity = match tls.cert_path() {
-                Some(path) => {
-                    info!("loading identity from {}", path.display());
-                    ServerCertificate::from_file(path)
-                        .with_context(|| ServerCertificateLoadError::File(path.to_path_buf()))?
-                }
-                None => {
-                    info!("downloading identity from edgelet");
-                    download_server_certificate()
-                        .await
-                        .with_context(|| ServerCertificateLoadError::Edgelet)?
-                }
+            let identity = if let Some(path) = tls.cert_path() {
```

Comment:
q: so, are we ok to override the cert via config? or should we always download from edgelet no matter what?

Reply:
by default (when not overwritten in config) it downloads cert from edgelet. If the customer really wants something different, they can provide their own cert. Also in dev mode it somewhat convenient to debug without edgelet running